src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Sin.cs

```diff
@@ -231,11 +231,17 @@ public static Vector128<double> Invoke(Vector128<double> x)
                     return ApplyScalar<SinOperatorDouble>(x);
                 }
 
+                // dn = |x| * (1 / π)
                 Vector128<double> almHuge = Vector128.Create(AlmHuge);
                 Vector128<double> dn = MultiplyAddEstimateOperator<double>.Invoke(uxMasked, Vector128.Create(1 / double.Pi), almHuge);
                 Vector128<ulong> odd = dn.AsUInt64() << 63;
                 dn -= almHuge;
-                Vector128<double> f = uxMasked - (dn * Vector128.Create(double.Pi)) - (dn * Vector128.Create(Pi_Tail1)) - (dn * Vector128.Create(Pi_Tail2));
+
+                // f = |x| - (dn * π)
+                Vector128<double> f = uxMasked;
+                f = MultiplyAddEstimateOperator<double>.Invoke(dn, Vector128.Create(-double.Pi), f);
+                f = MultiplyAddEstimateOperator<double>.Invoke(dn, Vector128.Create(-Pi_Tail1), f);
+                f = MultiplyAddEstimateOperator<double>.Invoke(dn, Vector128.Create(-Pi_Tail2), f);
```

Comment:
Were these changes to use MultiplyAddEstimateOperator opportunistic or necessary for precision? Interestingly I hadn't initially used this or FusedMultiplyAdd anywhere, and added it in because "why not", but your including it here suggests to me that maybe the precision in the original amd code was relying on gcc to automatically used fused-multiply add?

Reply:
Then should we be using FusedMultiplyAdd throughout rather than MultiplyAddEstimateOperator? The former is supposed to always use FMA, even if it's more expensive to do so, whereas the latter is free to fall back to non-FMA if it's faster to do so.

Reply:
Good point, I think we should do that although presumably we should be replacing every use of `MultiplyAddEstimateOperator` in these algorithms?

Reply:
>> Then should we be using FusedMultiplyAddOperator throughout rather than MultiplyAddEstimateOperator?

> although presumably we should be replacing every use of MultiplyAddEstimateOperator in these algorithms

Yes, that's what I meant by "throughout" :smile:

Reply:
I was wondering if we could do that in a follow-up, given that this change specifically fixes precision for the trig methods.

Reply:
Sure.